docs/implementation.md

```diff
@@ -32,3 +32,14 @@ There are two primary entry points for middleware:
 
 1. `robot.listenerMiddleware` - registers a new piece of middleware in a global array
 2. `middleware.execute` - executes all registered middleware in order
+
+## Persistence
+
+Hubot has an in-memory key-value store exposed as `robot.brain` that can be used to store and retrieve data.
```

Comment:
Changed it :smiley: 
I actually copied this first line from [`scripting.md`](https://github.com/github/hubot/blob/master/docs/scripting.md#persistence), so maybe I should update it too? I'm wondering if we should explain in `scripting.md` that `get` and `set` are only needed for `hubot-redis-brain`. Maybe in another pull request?